modules/serverless-apiserversource.adoc

```diff
@@ -0,0 +1,120 @@
+// Module included in the following assemblies:
+//
+// serverless/knative_eventing/serverless-kn-source.adoc
+
+[id="serverless-apiserversource_context"]
+= Using the API server source
+
+The API server source can be used to connect an event sink (for example, a Knative service) to the Kubernetes API server. The `ApiServerSource` watches for Kubernetes events and forwards them to the Knative Eventing broker.
+
+.Example ApiServerSource YAML
+
+----
+apiVersion: sources.knative.dev/v1alpha1
+kind: ApiServerSource
+metadata:
+  name: testevents
+  namespace: default
+spec:
+  serviceAccountName: events-sa
+  mode: Resource
+  resources:
+    - apiVersion: v1
+      kind: Event
+  sink:
+    ref:
+      apiVersion: eventing.knative.dev/v1alpha1
+      kind: Broker
+      name: default
+----
+
+== Prerequisites
+* Using the ApiServerSource requires you to have Knative Serving and Eventing installed.
+// For more information, see the documentation on xref:../.../serverless/installing-openshift-serverless.adoc#installing-openshift-serverless[Installing OpenShift Serverless].
+
+== Procedure
+// . Create the `default` broker in your namespace. For more information, see the documentation on xref:../.../serverless/knative_eventing/serverless-broker.adoc[Creating the broker].
+. Create a Service Account that the `ApiServerSource` runs as. Create a file named `serviceaccount.yaml` and copy the sample code into it:
++
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: events-sa
+  namespace: default
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: event-watcher
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - events
+    verbs:
+      - get
+      - list
+      - watch
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: k8s-ra-event-watcher
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: event-watcher
+subjects:
+  - kind: ServiceAccount
+    name: events-sa
+    namespace: default
+----
++
+If you want to re-use an existing Service Account with the appropriate permissions, you need to modify the `serviceaccount.yaml`.
+
+To create the service account from `serviceaccount.yaml`:
++
+----
+$ oc apply --filename serviceaccount.yaml
+----
+
+. Create a file named `k8s-events.yaml` and copy sample code into it:
++
+----
+apiVersion: sources.knative.dev/v1alpha1
+kind: ApiServerSource
+metadata:
+  name: testevents
+  namespace: default
+spec:
+  serviceAccountName: events-sa
+  mode: Resource
+  resources:
+    - apiVersion: v1
+      kind: Event
+  sink:
+    ref:
+      apiVersion: eventing.knative.dev/v1alpha1
```

Comment:
in TP / 0.13 the broker is in a different version, change the `apiVersion` to `eventing.knative.dev/v1beta1`

Reply:
Thanks!